From ad8e249fb9749556cf345ec6c431b96ba6cd5b27 Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
Date: Mon, 18 Nov 2024 17:14:00 +0200
Subject: [PATCH] bug_71489.at: Test removing option for OpenBSD

---
 tests/bug_71489.at | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/bug_71489.at b/tests/bug_71489.at
index 08196966..f541fecc 100644
--- a/tests/bug_71489.at
+++ b/tests/bug_71489.at
@@ -388,7 +388,7 @@ echo "Building local copy of the project"
       build_local_dir=$(pwd)
       AT_CHECK(
       [
-        $LIBTOOL --reorder-cache="$build_local_dir/liba/.libs:$build_local_dir/libb/.libs"
+        PATH="$PATH:/sbin" ldconfig -r
       ], 0, ignore, ignore);;
     *) ;;
     esac
-- 
2.45.2

