From 2865566ed2ed7a8a672d142f7b57776943d43b0b Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
Date: Fri, 4 Oct 2024 18:31:41 +0300
Subject: [PATCH] link-order2.at: Specify rpath for program link to dependency
 lib

---
 tests/link-order2.at | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/link-order2.at b/tests/link-order2.at
index af22af3e..8433ba2d 100644
--- a/tests/link-order2.at
+++ b/tests/link-order2.at
@@ -118,11 +118,11 @@ for type_of_depdepl in libtool non-libtool; do
        static=-all-static
     $LIBTOOL --mode=link $CC $CFLAGS $LDFLAGS $static -o liba1.la a1.lo -rpath $libdir
     $LIBTOOL --mode=link $CC $CFLAGS $LDFLAGS $static -o libb.la b.lo liba1.la -rpath $libdir
-    AT_CHECK([$LIBTOOL --mode=link $CC $CFLAGS $LDFLAGS $static -o main$EXEEXT main.$OBJEXT libb.la -la0],
+    AT_CHECK([$LIBTOOL --mode=link $CC $CFLAGS $LDFLAGS $static -o main$EXEEXT main.$OBJEXT libb.la -la0 -rpath $libdir],
 	     [], [ignore], [ignore])
     LT_AT_EXEC_CHECK([./main])
     # Now test that if we reverse the link order, the program fails.
-    AT_CHECK([$LIBTOOL --mode=link $CC $CFLAGS $LDFLAGS $static -o wrong$EXEEXT main.$OBJEXT -la0 libb.la],
+    AT_CHECK([$LIBTOOL --mode=link $CC $CFLAGS $LDFLAGS $static -o wrong$EXEEXT main.$OBJEXT -la0 libb.la -rpath $libdir],
 	     [], [ignore], [ignore])
     if test yes, != "$shared_fails,$static"; then
       LT_AT_EXEC_CHECK([./wrong], [1], [], [ignore], [|| (exit 1)])
-- 
2.45.2

