From 5bd6447ef5b3998666b4f955c89df31604d9b627 Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
Date: Mon, 11 Nov 2024 22:36:21 +0200
Subject: [PATCH] Add temporary quick check

---
 tests/bug_71489.at | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/tests/bug_71489.at b/tests/bug_71489.at
index ef4f51f7..08196966 100644
--- a/tests/bug_71489.at
+++ b/tests/bug_71489.at
@@ -382,6 +382,16 @@ echo "Building local copy of the project"
     cd build_local
     LT_AT_CONFIGURE([--prefix=$prefix], [$ltb2/configure])
     LT_AT_MAKE([])
+
+    case $host_os in
+    openbsd*)
+      build_local_dir=$(pwd)
+      AT_CHECK(
+      [
+        $LIBTOOL --reorder-cache="$build_local_dir/liba/.libs:$build_local_dir/libb/.libs"
+      ], 0, ignore, ignore);;
+    *) ;;
+    esac
 )
 
 # Although we have installed ltb2, we still expect that we are using the local
-- 
2.45.2

