From 7064eea526ecda1ff26016be003d4dbc24a72388 Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
Date: Wed, 6 Nov 2024 21:35:36 +0200
Subject: [PATCH] libtool.m4: Remove OpenBSD finish_cmds

---
 m4/libtool.m4 | 1 -
 1 file changed, 1 deletion(-)

diff --git a/m4/libtool.m4 b/m4/libtool.m4
index 857e9c11..4ea3b140 100644
--- a/m4/libtool.m4
+++ b/m4/libtool.m4
@@ -3024,7 +3024,6 @@ openbsd*)
     need_version=yes
   fi
   library_names_spec='$libname$release$shared_ext$versuffix $libname$shared_ext$versuffix'
-  finish_cmds='PATH="\$PATH:/sbin" ldconfig -m $libdir'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=yes
   ;;
-- 
2.45.2

