From b314b045e41dbdf0f1540d8cefb59a7457193343 Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
Date: Fri, 5 Jul 2024 16:21:56 +0300
Subject: [PATCH] tests/tagdemo.at: prefix possibly causing issues

---
 tests/tagdemo.at | 2 --
 1 file changed, 2 deletions(-)

diff --git a/tests/tagdemo.at b/tests/tagdemo.at
index 4f0055a5..e66def20 100644
--- a/tests/tagdemo.at
+++ b/tests/tagdemo.at
@@ -327,8 +327,6 @@ cos (0.0) = 1
 barbaz::baz is ok!
 convenience is ok!
 ]])
-
-prefix=`pwd`/_inst
 ]) # _LT_SETUP
 
 
-- 
2.45.1

