From 702e4b1837d5c7d1d39573101ab6b9c3dc99f7c8 Mon Sep 17 00:00:00 2001
From: Alastair McKinstry <mckinstry@debian.org>
Date: Sat, 12 Jan 2019 13:15:51 +0200
Subject: [PATCH 9/9] ltmain.in: Add error message for unknown version type

---
 build-aux/ltmain.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/build-aux/ltmain.in b/build-aux/ltmain.in
index c5bdbe5d..4e2a63fd 100644
--- a/build-aux/ltmain.in
+++ b/build-aux/ltmain.in
@@ -7031,6 +7031,9 @@ func_mode_link ()
 	    revision=$number_minor
 	    lt_irix_increment=no
 	    ;;
+	  *)
+	    func_fatal_configuration "$modename: unknown library version type '$version_type'"
+	    ;;
 	  esac
 	  ;;
 	no)
-- 
2.45.2

